src/Components/SingleArticle.jsx

```diff
@@ -1,6 +1,26 @@
-
+import {useEffect, useState} from "react"
+import {useParams} from "react-router-dom"
+import { getArticleByID } from "../api"
+import ArticleCard from "./ArticleCard"
 
 function SingleArticle() {
+   
+    const [article, setArticle] = useState({})
+    const {article_id} = useParams()
+
+    useEffect(() => {
+        getArticleByID(article_id)
+        .then((response) => {
+            setArticle(response.data.article[0])
+        }, [article_id])
```

Comment:
This useEffect is in an infinite loop and needs addressing as the dependancy array is in the incorrect place and may not need anything in it.